c10/core/impl/PyInterpreter.h

```diff
@@ -148,7 +152,8 @@ struct C10_API PyInterpreter {
       strides_sig* strides,
       sizes_sig* sizes,
       sym_sizes_sig* sym_sizes,
-      layout_sig* layout)
+      layout_sig* layout,
+      CUDATraceFunctionWrapper* trace_cuda_functions)
```

Comment:
From what you told me you were forced to use a pointer here because the struct is only forward-declared, and if you were to include the header where it's defined you'd hit a dependency cycle. However if you put the struct's definition in this file as suggested that won't be the case anymore and you can just pass this object by value.

Reply:
That's true, thank you. Fixed.